cmd/utils/auth.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE file in the project root for license information.
+
+package utils
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+// Further details about authentication:
+// https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/azidentity
+func GetCredentials() (*azidentity.ChainedTokenCredential, error) {
+	azCLI, err := azidentity.NewAzureCLICredential(nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating default authentication chain: %w", err)
+	}
+
+	// Fallback if users didn't get already authenticated using the Azure CLI
+	inBrowser, err := azidentity.NewInteractiveBrowserCredential(nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating interactive authentication chain: %w", err)
+	}
+
+	// Methods will be tried in that specific order: (1) Azure CLI (2) Interactive
+	chain, err := azidentity.NewChainedTokenCredential([]azcore.TokenCredential{azCLI, inBrowser}, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating credential chain: %w", err)
+	}
```

Comment:
Is there any way we can store the retrieved Token to avoid the user needs to authenticate at every execution?

Reply:
I will create an issue to track this task. Maybe folks from SDK team can help us with this?

cc @juan-lee